controller_manager:
  ros__parameters:
    update_rate: 1000

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController
      

trajectory_controller:
  ros__parameters:
    joints:
      - left_hip_revolute_joint
      - left_knee_revolute_joint
      - left_ankle_revolute_joint
      - right_hip_revolute_joint
      - right_knee_revolute_joint
      - right_ankle_revolute_joint
    command_interfaces:
      - position
    state_interfaces:
      - position

    allow_nonzero_velocity_at_trajectory_end: false
    allow_partial_joints_goal: true
    interpolate_from_desired_state: false

    gains:
      left_hip_revolute_joint: {p: 16.0, i: 0.001, d: 5.7}
      left_knee_revolute_joint: {p: 24.0, i: 0.0001, d: 21.0}
      left_ankle_revolute_joint: {p: 5.0, i: 0.005, d: 2.5}
      right_hip_revolute_joint: {p: 16.0, i: 0.001, d: 5.7}
      right_knee_revolute_joint: {p: 24.0, i: 0.0001, d: 21.0}
      right_ankle_revolute_joint: {p: 5.0, i: 0.005, d: 2.5}

    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.05
      left_hip_revolute_joint: {trajectory: 10.1, goal: 10.0}
      left_knee_revolute_joint: {trajectory: 10.1, goal: 10.0}
      left_ankle_revolute_joint: {trajectory: 10.1, goal: 10.0}
      right_hip_revolute_joint: {trajectory: 10.1, goal: 10.0}
      right_knee_revolute_joint: {trajectory: 10.1, goal: 10.0}
      right_ankle_revolute_joint: {trajectory: 10.1, goal: 10.0}